addons/mail/static/src/discuss_app/channel_selector.js

```diff
@@ -42,9 +44,12 @@ export class ChannelSelector extends Component {
                     ["name", "ilike", cleanedTerm],
                 ];
                 const fields = ["name"];
-                const results = await this.orm.searchRead("discuss.channel", domain, fields, {
-                    limit: 10,
-                });
+                await this.asyncFunctionOnInput.process(() =>
+                    this.orm.searchRead("discuss.channel", domain, fields, {
+                        limit: 10,
+                    })
+                );
+                const results = this.asyncFunctionOnInput.results || [];
```

Comment:
In case of quick return, this will execute with the results of the previous call.
This should be made to not resolve, or resolve with a sentinel value that makes caller ignore the call when not applicable.